src/WallsComputation.cpp

```diff
@@ -37,7 +37,20 @@ WallsComputation::WallsComputation(const Settings& settings, const LayerIndex la
  */
 void WallsComputation::generateWalls(SliceLayerPart* part, SectionType section_type)
 {
-    size_t wall_count = settings_.get<size_t>("wall_line_count");
+    constexpr auto get_wall_count_setting_name = [](SliceLayerPart* part)
+    {
+        switch (part->wall_exposed)
+        {
+        case SliceLayerPart::WallExposedType::BOTTOM_0:
+            return "wall_line_count_0";
+        case SliceLayerPart::WallExposedType::TOP:
+            return "wall_line_count_top";
+        case SliceLayerPart::WallExposedType::SIDE_ONLY:
+            return "wall_line_count";
+        }
+    };
```

Comment:
I don't see the need for a function here, since it is called only once, could you just do the switch/case and store the result in a variable ?

Reply:
I thought I could save an assignment this way, but that optimization will probably be done by the compiler anyway. I'll change it.